src/app/athkar-focus.tsx

```diff
@@ -114,13 +123,43 @@ const AthkarFocusScreen = () => {
   const currentCount = progressItem?.currentCount || 0;
   const totalCount = progressItem?.totalCount || currentAthkar?.count || 1;
   const progressPercentage = (currentCount / totalCount) * 100;
+
+  // Sync with store's currentAthkarIndex changes (auto-move functionality)
+  useEffect(() => {
+    // Re-animate progress when index changes due to auto-move
+    if (currentAthkar) {
+      animatedProgress.value = withSpring(progressPercentage / 100, {
+        damping: 15,
+        stiffness: 150,
+        mass: 1,
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentAthkarIndex, currentAthkar, progressPercentage]);
+
   const isCompleted = progressItem?.completed || false;
 
   // Check if all athkars in current session are completed
   const allCompleted = currentProgress
```

Comment:
For some reason when I tested it (maybe first time setup?) athkar will be empty, and this will be true.

let's add `currentProcess.length > 0 &&`